index.html

```diff
@@ -276,18 +276,54 @@ <h2>Methods</h2>
               return a promise rejected with a <code>DOMException</code> object whose
               <code>name</code> attribute has the value <code>InvalidStateError</code>.
               </p></li>
+              <li><p>Let <var>options</var> be the method's first argument.</p></li>
+              <li><p>Let <var>deviceId</var> be <var>options</var><code>.deviceId</code>.</p></li>
               <li><p>Let <var>p</var> be a new promise.</p></li>
               <li>
                 <p>Run the following steps in parallel:</p>
                 <ol>
+                  <li><p>Let <var>descriptor</var> be a PermissionDescriptor named
+                    <code>"speaker-selection"</code>.</p>
+                  </li>
+                  <li>
+                    <p>If <var>descriptor</var>'s
+                    <a data-cite="PERMISSIONS/#permission-state"
+                      >permission state</a> is
+                    <a data-cite="PERMISSIONS/#dom-permissionstate-granted"
+                      ><code>"denied"</code></a>, reject
+                      <var>p</var> with a new <code>DOMException</code> whose
+                      <code>name</code> attribute has the value
+                    <code>NotAllowedError</code>, and abort these steps.</p>
+                  </li>
                   <li><p>Probe the User Agent for available audio output devices.</p></li>
-                  <li><p>If there is no audio output device, reject
-                  <var>p</var> with a new <code>DOMException</code> whose <code>name</code> attribute
-                  has the value <code>NotFoundError</code>.</p></li>
+                  <li>
+                    <p>If there is no audio output device, reject <var>p</var>
+                    with a new <code>DOMException</code> whose <code>name</code>
+                    attribute has the value <code>NotFoundError</code> and abort
+                    these steps.</p>
+                  </li>
+                  <li>
+                    <p class="fingerprint">If <var>deviceId</var> is not
+                    <code>""</code> and matches an id previously exposed by
+                    <code>selectAudioOutput</code> in an earlier browsing
+                    session, the user agent MAY decide, based on its previous
+                    decision of whether to persist this id or not for this set
+                    of origins, to run the following sub steps:</p>
+                    <ol>
+                      <li>
+                        <p>Let <var>device</var> be the device identified by
+                        <var>deviceId</var>, if available.</p>
+                      </li>
+                      <li><p>If <var>device</var> is available, resolve
+                      <var>p</var> with either <var>deviceId</var> or a freshly
+                      rotated device id for <var>device</var>, and abort the
```

Comment:
This is probably implementation dependent, but can you clarify when we would reuse the deviceId and when we would use a rotated device Id? In Safari implementation, we would probably either reuse the deviceId or prompt.
A few additional edge cases:
- If we call twice selectAudioOutput with the same old deviceId, and we end up in this 'reuse deviceId' case, I would expect the implementation to always return the same deviceId (either the old one if rotation did not happen, or the same new one), except if rotation happened in between. The term 'freshly' might be a bit ambiguous.
- If the deviceId is already exposed as part of enumerateDevices, I would expect all implementations of selectAudioOutput to just return the provided deviceId.